backend/src/main/java/com/now/naaga/player/domain/Player.java

```diff
@@ -50,14 +46,46 @@ public Player(final String nickname,
         this(null, nickname, totalScore, member, FALSE);
     }
 
-    public Player(final Long id, final String nickname, final Score totalScore, final Member member, final boolean deleted) {
+    public Player(final Long id,
+                  final String nickname,
+                  final Score totalScore,
+                  final Member member,
+                  final boolean deleted) {
+        validateNickname(nickname);
         this.id = id;
         this.nickname = nickname;
         this.totalScore = totalScore;
         this.member = member;
         this.deleted = deleted;
     }
 
+    public static Player create(final String nickname,
+                                final Score score,
+                                final Member member) {
+        final String modifiedNickname = modifyWithValidNickname(nickname);
+        return new Player(modifiedNickname, score, member);
+    }
+
+    private static String modifyWithValidNickname(final String nickname) {
+        final String modifiedNickname = nickname.replaceAll("[^가-힣a-zA-Z0-9\\s]+", "");
+        if (modifiedNickname.length() > 20) {
+            return modifiedNickname.substring(0, NICKNAME_MAX_SIZE);
+        }
+        return modifiedNickname;
+    }
+
+    public void editNickname(final String newNickname) {
+        validateNickname(newNickname);
+        this.nickname = newNickname;
+    }
+
+    private void validateNickname(final String nickname) {
+        final boolean isUnavailableNickname = Pattern.matches("^.*[^가-힣a-zA-Z0-9\\s]+.*$", nickname);
```

Comment:
P3:
<img width="706" alt="image" src="https://github.com/woowacourse-teams/2023-naaga/assets/100172683/e74e178c-fe7a-47ff-a669-d7e5b4200b47">

Pattern.matches() 메서드는 내부적으로 패턴을 컴파일하고 매칭을 수행하기 때문에 반복 호출 시에 매번 패턴을 컴파일한다고 합니다.
따라서 미리 컴파일한 정적 Pattern객체를 만들고, static상수로 재사용하는 것은 어떨까요?
```suggestion
       final static Pattern PATTERN = Pattern.compile("^.*[^가-힣a-zA-Z0-9\\s]+.*$"); // 상수
       
       ...
       
         final boolean isUnavailableNickname = pattern.matcher(nickname).find();
```

Reply:
그렇네요.
Pattern에 명시가 되어있는 내용이네요.
근데 pattern.matcher()를 해도 Matcher라는 비교하는 객체를 생성하네요. 제 생각엔 크게 차이는 없을 것 같기는 하나, 정규표현식을 포함해서 상수화 해놓는게 나을 것 같아서 적용했습니다!!